### PR TITLE
Extract identity to a separate module

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

File: sdk/core/core/src/az_http_policy.c [hunk 1]
```diff
@@ -64,17 +63,24 @@ AZ_NODISCARD az_result az_http_pipeline_policy_retry(
   return az_http_pipeline_nextpolicy(p_policies, hrb, response);
 }
 
+typedef AZ_NODISCARD az_result (
+    *_az_auth_func)(void * const data, az_http_request_builder * const hrb);
+
+typedef struct {
+  _az_auth_func _func;
+} _az_auth;
```
Comment (gilbertw) [inline sdk/core/core/src/az_http_policy.c]: I would expect auth to be part of identity, thus _az_identity_auth.

File: sdk/identity/identity/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+endif()
+
+target_include_directories (az_identity PUBLIC inc)
+
+target_link_libraries(az_identity PUBLIC az_core)
+
+# make sure that users can consume the project as a library.
+add_library (az::identity ALIAS az_identity)
+
+# add_executable (az_identity_test test/main.c)
```
Comment (gilbertw) [inline sdk/identity/identity/CMakeLists.txt]: Remove this commented section

File: sdk/identity/identity/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+cmake_minimum_required (VERSION 3.12)
+
+project (az_identity LANGUAGES C)
+
+set(CMAKE_C_STANDARD 99)
+
+add_library (
+  az_identity
+  src/az_client_secret_credential.c
```
Comment (gilbertw) [inline sdk/identity/identity/CMakeLists.txt]: **In a subsequent pr** all the identity types should be moved to az_identity_*

Comment (antkmsft) [reply sdk/identity/identity/CMakeLists.txt]: Thanks! Subsequent PR: https://github.com/Azure/azure-sdk-for-c/pull/275

**Pull request merged**